sklearn/metrics/_ranking.py

```diff
@@ -1419,3 +1420,114 @@ def ndcg_score(y_true, y_score, k=None, sample_weight=None, ignore_ties=False):
     _check_dcg_target_type(y_true)
     gain = _ndcg_sample_scores(y_true, y_score, k=k, ignore_ties=ignore_ties)
     return np.average(gain, weights=sample_weight)
+
+
+def top_k_accuracy_score(y_true, y_score, k=2, normalize=True,
+                         sample_weight=None):
+    """Top-k Accuracy classification score.
+
+    This metric computes the number of times where the correct label is among
+    the top `k` labels predicted (ranked by predicted scores). Note that the
+    multilabel case isn't covered here.
+
+    If `k = 1`, :func:`accuracy_score` is called instead.
+
+    Read more in the :ref:`User Guide <top_k_accuracy_score>`
+
+    Parameters
+    ----------
+    y_true : array-like of shape (n_samples,)
+        True labels. Expected to be a sequence of int in the range
```

Comment:
Ordinarily we would accept string labels. Why not here?

Reply:
Because we are working with scores. If we accept string labels, then we need to infer which label correspond to each column of `y_score`

Reply:
My bad, I didn't noticed we were supporting that. I'm going to do as in `roc_auc_score` and add a `labels` parameter

Reply:
As `type_of_target([1, 2])` returns `'binary'`, do we assume that for all sklearn metrics `y_true` must contains each label at least once?